File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        
        for(int i=0; i<queries.length; i++){
            int min = -1;
            for(int j=queries[i][0]; j <= queries[i][1]; j++) {
                if(arr[j] > queries[i][2]) {
                    if(min == -1) min = arr[j];
                    else if(min > arr[j]) min = arr[j];
                }
                answer[i] = min;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int[,] queries) {
        int length = queries.GetLength(0);
        int[] answer = new int[length];
        
        for(int i = 0; i < length; i++)
        {
            int min = -1;
            for(int j = queries[i, 0]; j <= queries[i, 1]; j++)
            {
                if(queries[i, 2] < arr[j])
                {
                    min = min == -1 ? arr[j] : Math.Min(min, arr[j]);
                }
            }
            answer[i] = min;
        }
        
        return answer;
    }
}